### ISSUE TITLE
Add option to ignore code in ``if TYPE_CHECKING``

### ISSUE THREAD

**Issue by GoldsteinE** [PyCQA/pylint#3285]
### Is your feature request related to a problem? Please describe
In typed code, this is common pattern:
```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from some_module import some_type
```
If `some_type` is defined in stub, it doesn't have to exist in runtime (e. g. `wsgiref.types` is only available in `.pyi` files). Pylint issues an error for this code, while it's perfectly valid.
Pylint isn't a type checker, so it should ignore all code in `if TYPE_CHECKING:`.

### Describe the solution you'd like
Add option like to ignore all errors or, probably, only import-related errors in `if TYPE_CHECKING:` blocks.

**Comment by PCManticore** [comment]
Thanks, agree we should take more hints for `TYPE_CHECKING`.

**Comment by dpinol** [comment]
I'm trying to preprocess the source code commenting our the imports guarded with TYPE_CHECKING, and then removing  types with https://github.com/abarker/strip-hints. Otherwise, pylint would complain of the guarded types used in type hints.
unfortunately strip-hints also removes the types of dataclass fields, and hence I'm writing my own preprocessor :-|

**Comment by ssbarnea** [comment]
That bug allone makes it very inconvenient to keep using pylint on types projects, to a point where I am even considering removing it. Any chance we can have someone a patch to address it?

**Comment by Pierre-Sassoulas** [comment]
This is a high priority issue but no one is working on it right now, if you're willing to contribute it could go faster 😄

**Comment by matusvalo** [comment]
Just a note: As a workaround, disable all checks for a block should help.

```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    # pylint: disable=all
    from some_module import some_type
```

**Comment by Pierre-Sassoulas** [comment]
Thanks for the clever trick @matusvalo ! For readers who also are would be implementer, this workaround would not work for circular imports for example, so we still need this feature very badly :)
